cpp/cmake/thirdparty/get_thrust.cmake

```diff
@@ -80,6 +80,6 @@ function(find_and_configure_thrust VERSION)
   endif()
 endfunction()
```

Comment:
@robertmaynard I looked at this but I don't know what parts of this file are "custom" and which parts are "standard." I can tell the last line is different but I don't know what to do with the `CMAKE_INSTALL_INCLUDEDIR` and `CMAKE_INSTALL_LIBDIR` suggestions. Would you be able to give me a pointer or make a suggestion on what should be changed here?

Reply:
It would look roughly like this ( not tested ):
```cmake
  # We currently require cuDF to always build with a custom version of thrust. This is needed so
  # that build times of of cudf are kept reasonable, without this CI builds of cudf will be killed
  # as some source file can take over 45 minutes to build
  #
  include(GNUInstallDirs)
  set(CMAKE_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/libcudf")
  set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_INCLUDEDIR}/libcudf")
  
  set(CPM_DOWNLOAD_ALL TRUE)
  rapids_cpm_find(
    Thrust ${VERSION}
    BUILD_EXPORT_SET cudf-exports
    INSTALL_EXPORT_SET cudf-exports
    CPM_ARGS
    GIT_REPOSITORY https://github.com/NVIDIA/thrust.git
    GIT_TAG ${VERSION}
    GIT_SHALLOW TRUE ${cpm_thrust_disconnect_update} PATCH_COMMAND patch --reject-file=- -p1 -N <
                ${CUDF_SOURCE_DIR}/cmake/thrust.patch || true
    OPTIONS "THRUST_ENABLE_INSTALL_RULES TRUE"
  )

  if(NOT TARGET cudf::Thrust)
    thrust_create_target(cudf::Thrust FROM_OPTIONS)
  endif()

  if(Thrust_SOURCE_DIR) # only install thrust when we have an in-source version
    # Store where CMake can find our custom Thrust install
    include("${rapids-cmake-dir}/export/find_package_root.cmake")
    rapids_export_find_package_root(
      INSTALL Thrust [=[${CMAKE_CURRENT_LIST_DIR}/../../../include/libcudf/cmake/thrust/]=] cudf-exports
    )
  endif()
  ```

Reply:
@robertmaynard I applied this change in eaa9ef58caf36526b804366a2199d845f48c4b07 but it looks like the Python CMake logic is broken now? It appears to be pulling Thrust from rapids-cmake, if I'm reading the logs correctly -- but I'm not confident on what is happening. Is this a change that requires https://github.com/rapidsai/rapids-cmake/pull/231 to work, or is there something wrong in this PR?

Reply:
The issue is thrust-config.cmake is failing to find the headers so it can deduce the version.
```
19:02:11 CMake Error at /opt/conda/envs/rapids/include/rapids/thrust/thrust/cmake/thrust-config-version.cmake:10 (file):
19:02:16   file failed to open for reading (No such file or directory):
19:02:16 
19:02:16     /workspace/.conda-bld/work/python/cudf/_THRUST_VERSION_INCLUDE_DIR-NOTFOUND/thrust/version.h
19:02:16 Call Stack (most recent call first):
```

Can you try updating these values:
```
set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/libcudf")
...
rapids_export_find_package_root(
  INSTALL Thrust [=[${CMAKE_CURRENT_LIST_DIR}/../../../lib/libcudf/cmake/thrust/]=] cudf-exports
)
```

Reply:
If that doesn't work we can look at a local copy to debug.

Reply:
@robertmaynard d2b1456c4311ac4d7783ad912f9e3e92d6b72bf8 is still failing. I can't get the failures to reproduce locally, and I don't know what else to try here. Can we revert this suggestion and go back to 7c621d02301745ebae45e60d1a417127dd4ef337 where CI passed, then merge this PR?